AWS RDS cluster Table is added

laukik-uptycs opened the pull request:
By submitting a PR to this repository, you agree to the terms within the [Code of Conduct](https://github.com/Uptycs/cloudquery/blob/master/CODE_OF_CONDUCT.md). Please see the [contributing guidelines](https://github.com/Uptycs/cloudquery/blob/master/CONTRIBUTING.md) for how to create and submit a PR for this repo.

### Description

Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

Include any links supporting this change such as:

- GitHub Issue/PR number addressed or fixed
- Related pull requests/issues from other repos

### Testing

Describe the testing done. Setup required for testing. Details of unit or other tests added as a part of the PR.

alende-uptycs commented on extension/aws/rds/aws_rds_cluster.go:
```diff
@@ -0,0 +1,313 @@
+				"tableName": "aws_rds_cluster",
+				"account":   accountId,
+				"region":    *region.RegionName,
+				"task":      "ListCertificates",
```
DescribeDBClusters

alende-uptycs commented on extension/aws/rds/aws_rds_cluster.go:
```diff
@@ -0,0 +1,313 @@
+				"tableName": "aws_rds_cluster",
+				"account":   accountId,
+				"region":    *region.RegionName,
+				"task":      "ListCertificates",
```
DescribeDBClusters

alende-uptycs commented on extension/aws/rds/table_config.json:
```diff
@@ -0,0 +1,1002 @@
+        {
+          "sourceName": "DBClusters_AllocatedStorage",
+          "targetName": "allocated_storage",
+          "targetType": "BIGINT",
```
table.BigIntColumn in go file instead of TextColumn

alende-uptycs commented on extension/aws/rds/table_config.json:
```diff
@@ -0,0 +1,1002 @@
+        {
+          "sourceName": "DBClusters_BacktrackConsumedChangeRecords",
+          "targetName": "backtrack_consumed_change_records",
+          "targetType": "BIGINT",
```
table.BigIntColumn in go file instead of TextColumn

alende-uptycs commented on extension/aws/rds/table_config.json:
```diff
@@ -0,0 +1,1002 @@
+        {
+          "sourceName": "DBClusters_BacktrackWindow",
+          "targetName": "backtrack_window",
+          "targetType": "BIGINT",
```
****

alende-uptycs commented on extension/aws/rds/table_config.json:
```diff
@@ -0,0 +1,1002 @@
+        {
+          "sourceName": "DBClusters_BackupRetentionPeriod",
+          "targetName": "backup_retention_period",
+          "targetType": "BIGINT",
```
****

alende-uptycs commented on extension/aws/rds/table_config.json:
```diff
@@ -0,0 +1,1002 @@
+        {
+          "sourceName": "DBClusters_Capacity",
+          "targetName": "capacity",
+          "targetType": "BIGINT",
```
table.BigIntColumn in go file instead of TextColumn

alende-uptycs commented on extension/aws/rds/table_config.json:
```diff
@@ -0,0 +1,1002 @@
+        {
+          "sourceName": "DBClusters_ClusterCreateTime_loc_cacheZone_offset",
+          "targetName": "cluster_create_time_loc_cache_zone_offset",
+          "targetType": "INTEGER",
```
table.IntegerColumn in go file instead of TextColumn

alende-uptycs commented on extension/aws/rds/table_config.json:
```diff
@@ -0,0 +1,1002 @@
+        {
+          "sourceName": "DBClusters_ClusterCreateTime_ext",
+          "targetName": "cluster_create_time_ext",
+          "targetType": "BIGINT",
```
Change the column type in go file for all BIGINT attributes

alende-uptycs commented on utilities/utilities.go:
```diff
@@ -49,6 +49,7 @@ func ReadTableConfig(jsonEncoded []byte) error {
 		config.initParsedAttributeConfigMap()
 		TableConfigurationMap[tableName] = config
 	}
+	
```
let's unstage this unrequire file change

alende-uptycs commented:
Also, update `Dockerfile` and `extension/aws/tables.md`

laukik-uptycs commented:
**Description**

I have added the _AWS RDS Cluster_ information 

1. Added aws_rds_cluster.go file 
2. Created the necessary table-config.json file and enabled all first-level attributes
3. added those enabled columns in the table
4. Added the table name in table.md file
5. Added RDS table info in the docker file

**References**
https://docs.aws.amazon.com/cli/latest/reference/rds/describe-db-clusters.html

**Testing**
Tested using OSQuery

laukik-uptycs commented:
I have done the necessary changes please review and do the needful

alende-uptycs commented on extension/aws/rds/aws_rds_cluster.go:
```diff
@@ -0,0 +1,313 @@
+		// table.TextColumn("cluster_create_time_loc_cache_zone"),
+		// table.TextColumn("cluster_create_time_loc_cache_zone_is_dst"),
+		// table.TextColumn("cluster_create_time_loc_cache_zone_name"),
+		// table.TextColumn("cluster_create_time_loc_cache_zone_offset"),
```
Change the integer column data types as well wherever needed
